cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt

```diff
@@ -527,29 +528,43 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
         val (scope, _) = ctx.scopeManager.extractScope(source, source.scope)
         result.actualStartScope = scope ?: source.scope
 
-        // If the function does not allow function overloading, and we have multiple candidate
-        // symbols, the result is "problematic"
-        if (source.language !is HasFunctionOverloading && result.candidateFunctions.size > 1) {
-            result.success = PROBLEMATIC
-        }
-
-        // Filter functions that match the signature of our call, either directly or with casts;
-        // those functions are "viable". Take default arguments into account if the language has
-        // them.
-        result.signatureResults =
-            result.candidateFunctions
-                .map {
-                    Pair(
-                        it,
-                        it.matchesSignature(
-                            arguments.map(Expression::type),
-                            arguments,
-                            source.language is HasDefaultArguments,
+        // Resolution depends on language features
+        if (source.language !is HasFunctionOverloading) {
+            // If the function does not allow function overloading, and we have multiple candidate
+            // symbols, the result is "problematic"
+            if (result.candidateFunctions.size > 1) {
+                result.success = CallResolutionResult.SuccessKind.PROBLEMATIC
+            } else
+            // If we have only one candidate function and the number of arguments match, we can take
```

Comment:
This has multiple issues
1) This does not only affect C but other languages as well.
2) It does not take into account if languages have default arguments

At the very least I think we need to make this configurable. Depending on the use case you MAY or MAY NOT want to do this. For example if you actually want to check if something is resolving similar to a compiler (e.g., also inference turned off), you want to have an empty list of invokes edges here.

If you want to be very optimistic about matching types, this may be another (maybe even global?) configuration option. You can always of course override the `tryCast` in a language that depends on the C language if you want to have a "matches-everything-C-language`.

So I am very skeptical about this change.